Sources/Brave/Utility/AppStorageDebugComposer.swift

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import os.log
+
+struct AppStorageDebugComposer {
+  /// This toggle tells us whether the user attempted to clear browsing data.
+  /// This way we can ask the user to clear the data first
+  /// so the debug info shared with us will clearly show any outstanding big sized folders.
+  static var clearedDataThisSession = false
+  
+  /// This function prepares data to help us identify any app storage problems users may have.
+  static func compose() -> String {
+    var printDeviceInfo: String {
+      let device = UIDevice.current
+      let model = device.userInterfaceIdiom == .pad ? "iPad" : "iPhone"
+      let iOSVersion = "\(device.systemName) \(device.systemVersion)"
+      let appVersion = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String ?? "-"
+      
+      return "Setup: \(model), \(iOSVersion), Brave \(appVersion)"
+    }
+    
+    var printFolderTreeStructure: String {
+        let fm = FileManager.default
+        guard let enumerator = fm.enumerator(
+          at: URL(fileURLWithPath: NSHomeDirectory()),
+          includingPropertiesForKeys: nil
+        ) else { return "" }
+        
+        let formatter = ByteCountFormatter().then {
+          $0.countStyle = .file
+          $0.allowsNonnumericFormatting = false
+          $0.allowedUnits = [.useKB, .useMB, .useGB, .useTB]
+        }
+        
+        var result = ""
+        
+        while let file = enumerator.nextObject() as? URL {
+          do {
+            
+            let isDirectory = (try file.resourceValues(forKeys: [.isDirectoryKey])).isDirectory == true
+            // Skip individual files, skip folders below 4th level of nesting.
+            guard isDirectory, enumerator.level <= 4 else { continue }
+
+            let _1MB = 1000 * 1000
+            // Skip folders smaller than 1MB
+            guard let directorySize = try fm.directorySize(at: file), directorySize > _1MB else { continue }
+            
+            let formattedSize = formatter.string(fromByteCount: Int64(directorySize))
+            
+            let indentation = String(repeating: "·", count: (enumerator.level - 1) * 2)
```

Comment:
Can maybe just use a tab (`\t`) instead of this dot? (and dump * 2 in count)

Example result:

```
Setup: iPhone, iOS 16.2, Brave 1.46
Cleared Data: No
Library(277.7 MB)
	Application Support(140 MB)
		ads(16.8 MB)
		cmf-data(3.8 MB)
		Chromium(111.8 MB)
			BrowserMetrics(100.7 MB)
			BraveWallet(7.1 MB)
			cffkpbalmllkdoenhmdmpbkajipdjfam(3.4 MB)
		brave-today(4.4 MB)
	WebKit(6.3 MB)
		brave.kyle-arm64.local.id(6.3 MB)
			WebsiteData(2.6 MB)
			ContentRuleLists(3.7 MB)
	Caches(131 MB)
		com.hackemist.SDImageCache(8.8 MB)
			default(8.8 MB)
		brave.kyle-arm64.local.id(115.6 MB)
			WebKit(114.8 MB)
		Chromium(6.1 MB)
			Default(6.1 MB)
```

Reply:
sounds good i considered both